build-logic-commons/build-scan/build.gradle.kts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    `kotlin-dsl`
+}
+
+description = "Provides a plugin that configures build scans in the Gradle build"
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+}
+
+group = "gradlebuild"
+
+dependencies {
+    implementation("com.gradle:gradle-enterprise-gradle-plugin:3.8")
```

Comment:
I think this should be a `compileOnly` dependency, right?

My problem here is that the version of the enterprise plugin is now hard coded in another place. So for updating it we need to change the version here, in `settings.gradle.kts` and in `build-logic/build-platform`. It would be great to only need to change one version. I wonder if it would help to have a platform project in build logic settings, so we could omit the version everywhere.

Reply:
Yes, I want to get rid of these version declarations everywhere. I'll create a card and do it in a followup PR.